### PR TITLE
51 signup

### LINKED ISSUE #51: Signup component UI
Use Login.vue for examples

- Should have two forms (password/private key) (see zeplin) with tabs switch
- Should be able to copy generated password from the input field to clipboard
- Should have validations: all fields are required, passwords/pins should match, account name shouldn't be used on bitshares

### PR DESCRIPTION
**Pull request by roma219**
#### What's this PR do?
Sign up functionality, also improved UI components (styles, validations, titles)
#### Where should the reviewer start?
src/views/Signup
#### How should this be manually tested?
Try creating an account
#### Any background context you want to provide?
Don't forget to pull the latest `dev` from `vuex-bitshares` repo
#### What are the relevant tickets?
closes #51 
#### Screenshots (if appropriate)
image
image
#### Questions:
n/a
### URL
http://staging-ui.trusty.apasia.tech:8080/51-signup

### CODE REVIEW

File: src/views/Signup/Signup.vue [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        isUnique,
+        minLength: minLength(4),
+        hasSpecialSymbol(value) {
+          const hasDog = value.indexOf('@') > -1
+          const hasLine = value.indexOf('-') > -1
+          const hasNumber = /\d/.test(value)
+          return hasDog || hasLine || hasNumber
+        },
+        noBadSymbolAtEnd(value) {
+          if (value.indexOf('@') === value.length - 1) return false
```
Comment (youaresofunny) [inline src/views/Signup/Signup.vue]: return (value.endsWith('@') || value.endsWith('-') || value.endsWith('.')) ? false : true;

File: src/views/Signup/Signup.vue [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import { utils, getUser } from 'vuex-bitshares/src/services/api/account.js'
+import dictionary from 'vuex-bitshares/test/brainkey_dictionary.js'
+import '@icons/copy'
+import '@icons/paste'
+
+const isUnique = (name) => {
+  if (name === '') return true
+  return new Promise(async(resolve, reject) => {
+    const resp = await getUser(name)
+    resolve(!resp.success)
```
Comment (youaresofunny) [inline src/views/Signup/Signup.vue:109]: This code looks strange

**Review by youaresofunny — CHANGES_REQUESTED** [review]
image

Can't understand here why my name is not valid

File: src/views/Signup/Signup.vue (hunk 2 above)
Comment (roma219) [reply src/views/Signup/Signup.vue:109]: Gonna make a comment for it. It's async vuelidate validation, you need to return a promise and after your sync action you need to resolve it with either true/false to signal it it's valid or not.

**Comment by roma219** [comment]
@youaresofunny "RE: name is not valid - it's just a response from the faucet, probably something was down at a time, just registered user with this username.

**Review by youaresofunny — APPROVED** [review]

**Comment by youaresofunny** [comment]
:+1:

**Pull request merged**